SpockOpaleye/appcore/src/Utils.hs

```diff
@@ -0,0 +1,10 @@
+module Utils where
+
+import UserDefs
+import CryptoDef
+import InternalUtils
+import Data.Text
+import TenantId
+
+fillPassword :: UserIncoming -> BcryptPassword -> UserPoly () () () TenantId Text BcryptPassword (Maybe Text) (Maybe Text) ()
+fillPassword incUser hash = incUser { _userpolyPassword = hash }
```

Comment:
Here is the type class that defines the `password` lens.

```
class HasPassword s a | s -> a where
  password :: Control.Lens.Type.Lens' s a
  {-# MINIMAL password #-}
        -- Defined at /vagrant/projects/haskell-webapps/SpockOpaleye/appcore/src/Lenses.hs:18:3
instance HasPassword
           (UserPoly
              key0
              created_at0
              updated_at0
              tenant_id0
              username0
              password0
              firstname0
              lastname0
              status0)
           password0
```
Doesn't it mean that the password setter can only accept a value of type password0, when used with type,

```
 (UserPoly
              key0
              created_at0
              updated_at0
              tenant_id0
              username0
              password0
              firstname0
              lastname0
              status0)

```
Does the functional dependency prevents the above type from having an second instance that can accept another type for password field other than password0

And since IncomingUser has type 'Text' for password, it follows that the setter can only accept values of type Text.